compiler/rustc_ast/src/token.rs

```diff
@@ -590,6 +591,9 @@ impl Token {
     /// for which spans affect name resolution and edition checks.
     /// Note that keywords are also identifiers, so they should use this
     /// if they keep spans or perform edition checks.
+    //
+    // Note: `Parser::uninterpolated_token_span` may give better information
```

Comment:
What is the difference?
This span is significant, in the sense that it's not just used for diagnostics, but for things like edition checks as well.
Ideally we should have a single way to determine it.

Reply:
By the time `Nonterminal` is fully removed, `Token::uninterpolated_span` will only uninterpolate for `NtIdent`/`NtLifetime`. A lot of the time that's good enough. `Parser::uninterpolated_token_span` is required to uninterpolate anything else, because you have to be able to look ahead past the opening invisible literal.

It's possible to make `Parser::uninterpolated_token_span` the one way to do it, I will add a commit that does that.